plugins/redis/redisasync.cpp

```diff
@@ -0,0 +1,870 @@
+    return value;
+}
+ECL_REDIS_API signed __int64 ECL_REDIS_CALL AsyncRGetInt8(ICodeContext * ctx, const char * options, const char * key, unsigned __int64 database)
+{
+    signed __int64 value;
+    AsyncRGet(ctx, options, key, value, NULL, database);
+    return value;
+}
+ECL_REDIS_API unsigned __int64 ECL_REDIS_CALL AsyncRGetUint8(ICodeContext * ctx, const char * options, const char * key, unsigned __int64 database)
+{
+    unsigned __int64 value;
+    AsyncRGet(ctx, options, key, value, NULL, database);
+    return value;
+}
+ECL_REDIS_API void ECL_REDIS_CALL AsyncRGetStr(ICodeContext * ctx, size32_t & returnLength, char * & returnValue, const char * options, const char * key, unsigned __int64 database)
+{
+    size_t _returnLength;
+    AsyncRGet(ctx, options, key, _returnLength, returnValue, NULL, database);
+    returnLength = static_cast<size32_t>(_returnLength);
+}
+ECL_REDIS_API void ECL_REDIS_CALL AsyncRGetUChar(ICodeContext * ctx, size32_t & returnLength, UChar * & returnValue,  const char * options, const char * key, unsigned __int64 database)
+{
+    size_t returnSize = 0;
+    AsyncRGet(ctx, options, key, returnSize, returnValue, NULL, database);
+    returnLength = static_cast<size32_t>(returnSize/sizeof(UChar));
+}
+ECL_REDIS_API void ECL_REDIS_CALL AsyncRGetUtf8(ICodeContext * ctx, size32_t & returnLength, char * & returnValue, const char * options, const char * key, unsigned __int64 database)
+{
+    size_t returnSize = 0;
+    AsyncRGet(ctx, options, key, returnSize, returnValue, NULL, database);
+    returnLength = static_cast<size32_t>(rtlUtf8Length(returnSize, returnValue));
+}
+ECL_REDIS_API void ECL_REDIS_CALL AsyncRGetData(ICodeContext * ctx, size32_t & returnLength, void * & returnValue, const char * options, const char * key, unsigned __int64 database)
+{
+    size_t _returnLength = 0;
+    AsyncRGet(ctx, options, key, _returnLength, returnValue, NULL, database);
+    returnLength = static_cast<size32_t>(_returnLength);
+}
+//----------------------------------------LOCKING WRAPPERS------------------------------------------------------------------------------------------
+ECL_REDIS_API unsigned __int64 ECL_REDIS_CALL RGetLockObject(ICodeContext * ctx, const char * options, const char * key, unsigned __int64 database)
+{
+    Owned<KeyLock> lockObject;
+    StringBuffer channel;
+    channel.set(REDIS_LOCK_PREFIX);
+    channel.append("_").append(key).append("_");
+    channel.append("_").append(database).append("_");
+
+    //srand(unsigned int seed);
+    channel.append(rand()%REDIS_MAX_LOCKS+1);
+
+    lockObject.set(new KeyLock(ctx, options, key, channel.str(), database));
+    return reinterpret_cast<unsigned long long>(lockObject.get());
+}
+ECL_REDIS_API bool ECL_REDIS_CALL RMissThenLock(ICodeContext * ctx, unsigned __int64 _lockObject)
+{
+    KeyLock * lockObject = (KeyLock*)_lockObject;
+    const char * channel = lockObject->getChannel();
+    if (!lockObject || strlen(channel) == 0)
```

Comment:
Yep, I can see that originally strlen(channel) was lockObject->getChannel(). I then moved it out to use else where, without changing the null test, furthermore removing the second use altogether.